### PR TITLE
Simple integration test for create cluster

### PR DESCRIPTION
**Pull request by justinsb**
Calls create cluster command
Compares to expected YAML output

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1188)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Umm CI failed.

**Comment by justinsb** [comment]
CI failed because we were looking for `~/.ssh/id_rsa.pub` ... now creating it in the integration test.

**Comment by justinsb** [comment]
And travis is finally happy :-)

**Comment by chrislovecnm** [comment]
Does this depend on other work?

File: cmd/kops/integration_test.go [hunk 1]
```diff
@@ -67,39 +67,10 @@ func runTest(t *testing.T, clusterName string, srcDir string) {
 	factoryOptions := &util.FactoryOptions{}
```
Comment (chrislovecnm) [inline cmd/kops/integration_test.go]: So is this actually an integration test that will build a cluster? We have unit, integration, e2e. What do the names mean?

**Comment by justinsb** [comment]
They're probably not integration tests.  Maybe this is your opportunity to coin a new term and take it onto the testing speaker circuit ;-)

But ... the idea is that I have divided the system into 3 parts:

1) CLI -> Model (this test)
2) Model -> Terraform (Terraform being our task model, and then a bit more work on top)
3) Tasks -> Cloud (with mock AWS in cloudmock)

If those three pieces are all working, there's hopefully not a lot in between.

And then e2e is to catch where (3) in particular fails in the real world, due to rate limits, inaccuracies of the AWS mock etc.

**Comment by chrislovecnm** [comment]
And e2e flakes on is as well. :facepalm

**Comment by chrislovecnm** [comment]
LGTM

**Pull request merged**